### PR TITLE
Cache DiskArray write header in-memory

### PR DESCRIPTION
**Pull request by benjaminwinger**
Still a bunch of tests failing.

Additionally, as far as I can tell, the `InMemDiskArray` isn't actually doing anything meaningful. operator[] is unused (except in the `HashIndexBuilder` via `InMemDiskArrayBuilder`) and get/update are just using the `BaseDiskArray` implementation. The description suggests that it's supposed to cache the entirety of the metadataDA in memory, but I don't think it's done that for a while (hence our attempts to cache metadata locally when updating columns).

The cached header was creating some issues with the InMemDiskArray code, and while I think I was able to fix them in the initial version of this (or at least fix the crashes), being unused the code is not well tested. I've replaced InMemDiskArray with a thin shim for now, as it was doing some unnecessary work reading the pages into memory.

### CODE REVIEW

File: src/storage/store/var_list_column.cpp [hunk 1]
```diff
@@ -246,6 +246,11 @@ void VarListColumn::scanFiltered(Transaction* transaction, node_group_idx_t node
         listOffset += resultVector->getValue<list_entry_t>(pos).size;
     }
 }
```
Comment (ray6080) [inline src/storage/store/var_list_column.cpp:248]: Add an empty line here.

File: src/storage/store/node_table.cpp [hunk 2]
```diff
@@ -160,6 +160,15 @@ void NodeTable::prepareCommit(Transaction* transaction, LocalTable* localTable)
...
+    tableData->prepareCommit();
+    wal->addToUpdatedTables(tableID);
+}
+
+void NodeTable::prepareCommit() {
+    if (pkIndex) {
+        pkIndex->prepareCommit();
+    }
+    tableData->prepareCommit();
     wal->addToUpdatedTables(tableID);
```
Comment (ray6080) [inline src/storage/store/node_table.cpp:172]: I don't think you need L172.

File: src/storage/storage_manager.cpp [hunk 3]
```diff
@@ -127,6 +127,11 @@ void StorageManager::dropTable(table_id_t tableID) {
 
 void StorageManager::prepareCommit(transaction::Transaction* transaction) {
     transaction->getLocalStorage()->prepareCommit();
+    for (auto tableID : wal->getUpdatedTables()) {
```
Comment (ray6080) [inline src/storage/storage_manager.cpp:130]: Can you add a brief comment here to explain the changes? Better for readability.

**Review by ray6080 — APPROVED** [review]

File: src/storage/store/var_list_column.cpp (hunk 1 above)
Comment (benjaminwinger) [reply src/storage/store/var_list_column.cpp:248]: Hmm. apparently clang-format >=14 can enforce this: https://clang.llvm.org/docs/ClangFormatStyleOptions.html#separatedefinitionblocks. We should probably look at upgrading at some point.

**Pull request merged**